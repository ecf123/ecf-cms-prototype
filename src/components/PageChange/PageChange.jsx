import "./PageChange.scss";

import React from "react";

const PageChange = ({
  squareOne,
  squareTwo,
  squareThree,
  squareFour,
  squareFive,
  handlePageChange,
}) => {
  return (
    <section>
      <div className="pageChange">
        {/* add this to handlePageChange function - pageChange_button.classList.add("-active") */}
        <div className="pageChange_button">&lt;</div>
        <div className="pageChange_button" onClick={handlePageChange}>
          1{squareOne}
        </div>
        <div className="pageChange_button" onClick={handlePageChange}>
          2{squareTwo}
        </div>
        <div className="pageChange_button" onClick={handlePageChange}>
          3{squareThree}
        </div>
        <div className="pageChange_button" onClick={handlePageChange}>
          ...{squareFour}
        </div>
        <div className="pageChange_button" onClick={handlePageChange}>
          25{squareFive}
        </div>
        <div className="pageChange_button">&gt;</div>
      </div>
    </section>
  );
};

export default PageChange;
