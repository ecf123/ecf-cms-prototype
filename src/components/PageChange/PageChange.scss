@use "../../styles/variables"as *;
@use "../../styles/typography"as *;


section {
    position: absolute;
    width: 83.8rem;
    height: 3.75rem;
    left: 20.8rem;
    top: 50.3rem;
    display: flex;
    justify-content: flex-start;

    .pageChange {
        display: flex;
        justify-content: space-between;
        height: 2rem;
        width: 15.5rem;

    &_button {
        height: 2rem;
        width: 2rem;
        border: solid 1px $color-grey-normal;
        border-radius: 4px;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: $color-white-smoke;
        color: $color-grey-dark;
        font-size: $font-size-extra-small;
        font-weight: $font-weight-regular;


        &-active {
            background-color: $color-black;
            color: $color-white;
        }

        &-inactive {
            opacity: 0.5;
        }
    }

}

}


