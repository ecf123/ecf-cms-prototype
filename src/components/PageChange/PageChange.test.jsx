import {render, screen} from "@testing-library/react"
import { customRender } from "../../utils/testUtils";
import PageChange from "./PageChange";

it("should show the page number on screen", () => {
    //.1 Arrange
    render(<PageChange totalPages={20}/>)

    //.2 Act
    const pageNumber = screen.findByText(/2/i);
    const PageRight = screen.findByDisplayValue(">");

    //3. Assert
    expect(pageNumber).toBeTruthy();
    expect(PageRight).toBeTruthy();
}) 

it("Should render PageChange", () => {
    const { container } = customRender(
    <PageChange/>
    );

    expect(container).toMatchSnapshot();
});


it("Should render PageChange with props", () => {
    const { container } = customRender(
    <PageChange totalPages={20}/>
    );

    expect(container).toMatchSnapshot();
});
