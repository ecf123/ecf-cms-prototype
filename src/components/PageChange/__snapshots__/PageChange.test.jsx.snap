// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render PageChange 1`] = `
<div>
  <section>
    <div
      class="pageChange"
    >
      <div
        class="pageChange_button"
      >
        &lt;
      </div>
      <div
        class="pageChange_button"
      >
        1
      </div>
      <div
        class="pageChange_button"
      >
        2
      </div>
      <div
        class="pageChange_button"
      >
        3
      </div>
      <div
        class="pageChange_button"
      >
        ...
      </div>
      <div
        class="pageChange_button"
      >
        25
      </div>
      <div
        class="pageChange_button"
      >
        &gt;
      </div>
    </div>
  </section>
</div>
`;

exports[`Should render PageChange with props 1`] = `
<div>
  <section>
    <div
      class="pageChange"
    >
      <div
        class="pageChange_button"
      >
        &lt;
      </div>
      <div
        class="pageChange_button"
      >
        1
        250
      </div>
      <div
        class="pageChange_button"
      >
        2
      </div>
      <div
        class="pageChange_button"
      >
        3
      </div>
      <div
        class="pageChange_button"
      >
        ...
      </div>
      <div
        class="pageChange_button"
      >
        25
      </div>
      <div
        class="pageChange_button"
      >
        &gt;
      </div>
    </div>
  </section>
</div>
`;
